### ISSUE TITLE
String.replace is tracking "lastIndex" on non-global regexps?

### ISSUE THREAD

**Issue by otac0n** [fholm/IronJS#32]
This is blocking the V8 test `regexp.js`.

Here is the minimum repro case:

```
var re89 = /\s+$/;
'Oyvpxchaxg\x0a'.replace(re89, '');
'Svanamra\x0a'.replace(re89, '');
```

It doesn't appear that the `if not search.Global` and `search.Put` stuff is correct for String.prototype.replace.

`Native.String.fs lines 213-247`

**Comment by otac0n** [comment]
I'll take a look at this as soon as I can, but it may be a couple of days.  The spec says "Do the search in the same manner as in `String.prototype.match`, including the update of  `searchValue.lastIndex.`", but as far as I'm aware, `lastIndex` is only updated for global regexps.

**Comment by fholm** [comment]
Fixed in [63adef9ebaa354c](https://github.com/fholm/IronJS/commit/850dad883d2b5c75aefc5f82a63adef9ebaa354c)
